features/api-layer/common/src/main/java/org/opennms/features/apilayer/common/distributed/KeyValueStoreWrapper.java

```diff
@@ -0,0 +1,129 @@
+/*******************************************************************************
+ * This file is part of OpenNMS(R).
+ *
+ * Copyright (C) 2022 The OpenNMS Group, Inc.
+ * OpenNMS(R) is Copyright (C) 1999-2022 The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is a registered trademark of The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License,
+ * or (at your option) any later version.
+ *
+ * OpenNMS(R) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with OpenNMS(R).  If not, see:
+ *      http://www.gnu.org/licenses/
+ *
+ * For more information contact:
+ *     OpenNMS(R) Licensing <license@opennms.org>
+ *     http://www.opennms.org/
+ *     http://www.opennms.com/
+ *******************************************************************************/
+
+package org.opennms.features.apilayer.common.distributed;
+
+import java.util.Map;
+import java.util.Optional;
+import java.util.OptionalLong;
+import java.util.concurrent.CompletableFuture;
+
+import org.opennms.integration.api.v1.distributed.KeyValueStore;
+
+public class KeyValueStoreWrapper<T> implements KeyValueStore<T> {
+    private final org.opennms.features.distributed.kvstore.api.KeyValueStore store;
+
+    public KeyValueStoreWrapper(org.opennms.features.distributed.kvstore.api.KeyValueStore store) {
+        this.store = store;
+    }
+
+    @Override
+    public long put(String key, T value, String context) {
+        return store.put(key, value, context);
```

Comment:
I think we should add a prefix to the context which would limit what the OIA plugins can touch/access - at worse they could access keys for other plugins.

Reply:
Good idea